persistence/trees/trees.go

```diff
@@ -88,6 +99,17 @@ type treeStore struct {
 	treeStoreDir string
 	rootTree     *stateTree
 	merkleTrees  map[string]*stateTree
+
+	// PrevState holds a previous view of the Worldstate.
+	// The tree store rolls back to this view if errors are encountered during block application.
+	PrevState *Worldstate
+}
+
+// Worldstate holds a (de)serializable view of the entire tree state.
+type Worldstate struct {
```

Comment:
Does this need to be publically exposed? 

Since we'll have a CLI for it, might be worth making it local and adding a TODO (with the cli ticket) to add an interface for it?